File: src/main/java/com/polygloat/dtos/response/UserResponseDTO.java
package com.polygloat.dtos.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserResponseDTO {

    private Long id;

    private String name;

    private String username;

    public static UserResponseDTO fromEntity(com.polygloat.model.UserAccount user) {
        return UserResponseDTO.builder().username(user.getUsername()).name(user.getName()).id(user.getId()).build();
    }
}

File: backend/core/src/main/kotlin/io/tolgee/dtos/response/UserResponseDTO.kt
package io.tolgee.dtos.response

import io.tolgee.model.UserAccount

data class UserResponseDTO(
  var id: Long? = null,
  var name: String? = null,
  var username: String? = null,
  var emailAwaitingVerification: String? = null,
) {
  companion object {
    fun fromEntity(userAccount: UserAccount): UserResponseDTO {
      return UserResponseDTO(
        name = userAccount.name,
        username = userAccount.username,
        id = userAccount.id,
        emailAwaitingVerification = userAccount.emailVerification?.newEmail,
      )
    }
  }
}
